### PR TITLE
Update cardano-ledger dependency

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

**Comment by newhoggy** [comment]
This fails with:

```
src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:564:51: error:
    • Could not deduce: Data.Sharing.Share (LC.TxOut era)
                        ~ Data.Sharing.Interns
                            (SL.Credential 'SL.Staking (Cardano.Ledger.Era.Crypto era))
        arising from a use of ‘fromCBOR’
      from the context: ShelleyBasedEra era
        bound by the type signature for:
                   decodeShelleyResult :: forall era result.
                                          ShelleyBasedEra era =>
                                          BlockQuery (ShelleyBlock era) result
                                          -> forall s. Decoder s result
        at src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:(551,1)-(554,31)
      or from: result ~ SL.EpochState era
        bound by a pattern with constructor:
                   DebugEpochState :: forall era.
                                      BlockQuery (ShelleyBlock era) (SL.EpochState era),
                 in a case alternative
        at src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:564:5-19
    • In the expression: fromCBOR
      In a case alternative: DebugEpochState -> fromCBOR
      In the expression:
        case query of
          GetLedgerTip -> decodePoint decode
          GetEpochNo -> decode
          GetNonMyopicMemberRewards {} -> decode
          GetCurrentPParams -> fromCBOR
          GetProposedPParamsUpdates -> fromCBOR
          GetStakeDistribution -> fromCBOR
          GetUTxOByAddress {} -> fromCBOR
          GetUTxOWhole -> fromCBOR
          DebugEpochState -> fromCBOR
          GetCBOR {} -> decode
          GetFilteredDelegationsAndRewardAccounts {} -> fromCBOR
          GetGenesisConfig -> fromCBOR
          DebugNewEpochState -> fromCBOR
          DebugChainDepState -> fromCBOR
          GetRewardProvenance -> fromCBOR
          GetUTxOByTxIn {} -> fromCBOR
          GetStakePools -> fromCBOR
          GetStakePoolParams {} -> fromCBOR
          GetRewardInfoPools -> fromCBOR
    • Relevant bindings include
        query :: BlockQuery (ShelleyBlock era) result
          (bound at src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:555:21)
        decodeShelleyResult :: BlockQuery (ShelleyBlock era) result
                               -> forall s. Decoder s result
          (bound at src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:555:1)
    |
564 |     DebugEpochState                            -> fromCBOR
```

If I delete the line:

```
DebugEpochState                            -> fromCBOR
```

The project compiles, so this is the only remaining compiler error to fix.

The demand for the constraint comes from here:

https://github.com/input-output-hk/cardano-ledger/blob/master/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState.hs#L507

File: cabal.project [hunk 1]
```diff
@@ -230,6 +233,8 @@ source-repository-package
     libs/cardano-ledger-core
     libs/cardano-ledger-pretty
     libs/cardano-protocol-tpraos
+    libs/compact-map
+    libs/ledger-state
```
Comment (lehins) [inline cabal.project:237]: You shouldn't need to add this package, it is only used internally by ledger for memory benchmarking:
```suggestion
```

**Review by lehins — CHANGES_REQUESTED** [review]
This PR should fix the current build error: https://github.com/input-output-hk/cardano-ledger/pull/2577

File: cabal.project (hunk 1 above)
Comment (newhoggy) [reply cabal.project]: Removed.  Thanks!

**Comment by nfrisby** [comment]
The current CI failure suggests the Shelley ledger state serialization has changed. @lehins @JaredCorduan @nc6 Is that expected here? Thanks.

File: cabal.project [hunk 2]
```diff
@@ -1,4 +1,4 @@
-index-state: 2021-02-15T00:00:00Z
+index-state: 2021-08-17T00:00:00Z
```
Comment (nfrisby) [inline cabal.project:1]: The new `cardano-ledger-specs` commit you're pointing to (4595b73a693d0a78f05ecbcbf631caa568dde5d8) uses `index-state: 2021-12-03T06:07:23Z`.

My intuition is that we'll benefit from having the same index-state as our primary upstream library. Would that work here?

**Comment by lehins** [comment]
@nfrisby It has changed indeed in https://github.com/input-output-hk/cardano-ledger/pull/2557. In particular the order in which fields are being serialized/deserialized has changed for `DPState`  and `EpochState`

**Comment by nfrisby** [comment]
Is this PR still live, @newhoggy ?

File: cabal.project (hunk 2 above)
Comment (newhoggy) [reply cabal.project]: I switched to `2021-12-03T06:07:23Z` and the build looks fine.

Comment (newhoggy) [reply cabal.project:1]: Okay, I switched it back.  I remember why now.  The way `nix` is set up, I cannot push it beyond this date.

**Comment by newhoggy** [comment]
Do not Merge this PR until the release.

**Comment by newhoggy** [comment]
This PR is no longer needed.

**Pull request closed without merging**